setup.py

```diff
@@ -20,24 +20,27 @@
 django_deps = ["django>=2.2.9,<4.0"]
 elastic_deps = ["elasticsearch-dsl>=6.4,<8.0"]
 mongo_deps = ["pymongo~=3.11", "mongomock~=3.22"]
-server_deps = ["uvicorn~=0.13.3", "Jinja2~=2.11"] + mongo_deps
+server_deps = ["uvicorn~=0.13.4", "Jinja2~=2.11"] + mongo_deps
 
 # Client minded
 aiida_deps = ["aiida-core~=1.5.2"]
 ase_deps = ["ase~=3.21"]
 cif_deps = ["numpy~=1.19"]  # Keep at 1.19 for Python 3.6 support
 pdb_deps = cif_deps
-pymatgen_deps = ["pymatgen==2021.2.8.1"]
-jarvis_deps = ["jarvis-tools==2020.11.27"]
+pymatgen_deps = [
+    'pymatgen==2021.2.8.1;python_version<"3.7"',
+    'pymatgen==2021.2.16;python_version>"3.6"',
+]
+jarvis_deps = ["jarvis-tools==2021.2.22"]
```

Comment:
I think I've lost sight of how we're meant to be handling these calendar versioned packages; should we not specify as loose a version as possible in the setup.py, then test with the versions in the requirements? e.g. `pymatgen>2020,<2021.2.8.1;python_version<"3.7"?

Also, are these versions even the right way around? The conditions on `python_version` seem to overlap... Should it be 

```suggestion
pymatgen_deps = [
    'pymatgen==2021.2.8.1;python_version<="3.6"',
    'pymatgen==2021.2.16;python_version>"3.6"',
]
jarvis_deps = ["jarvis-tools==2021.2.22"]
```

instead?

Reply:
Ah right! Yeah, I wanted to do it similarly to the `requirements` file. But the way I have it now is definitely wrong :)

Reply:
Ah no. The packages actually don't overlap. I did it right the first time around. So if the python version is up to (but not including) v3.7 it will install the older version. While if the python version is later than (but not including) v3.6 it will install the latest version.

Reply:
The change you're suggesting will result in the same as what is currently used.

Reply:
> I think I've lost sight of how we're meant to be handling these calendar versioned packages; should we not specify as loose a version as possible in the setup.py, then test with the versions in the requirements? e.g. `pymatgen>2020,<2021.2.8.1;python_version<"3.7"?

I believe the reason why we are strict for the calendar versioned packages, is due to the fact that we cannot ensure we correctly support its sub-dependencies, together with any major backwards-incompatible changes that could occur. We can ensure this support and compatibility for semantically versioned packages (by keeping below some version number), but this is not possible for the calendar versioned packages.

This was the solution. Be strict :)

Reply:
Hmm okay, could you just add a comment that pymatgen<whatever does not support Python 3.6? Otherwise happy to continue as we are, if that is still preferable

Reply:
This should be clear from the dependencies though? v2021.2.8.1 is the latest version to support Python 3.6. All newer versions does not support Python 3.6 (since after that `numpy` has been fixed to v1.20.1).

Reply:
There is no way to make a range, lower or higher limit, unfortunately, since they don't use semantic versioning.

Reply:
Okay fine, starting to think we should just drop 3.6 support too really, but that's for another day.